glean-core/metrics.yaml

```diff
@@ -75,6 +75,26 @@ glean.internal.metrics:
       - glean-team@mozilla.com
     expires: never
 
+  windows_build_number:
+    type: quantity
+    lifetime: application
+    send_in_pings:
+      - glean_client_info
+    description: |
+      The optional user-visible version of the Windows Build Number
+      (e.g. "22000"). If the version detection fails, this metric gets set
+      to `Unknown`.
```

Comment:
`quantity` doesn't support a value of `Unknown`, though?

Reply:
Ah, good point, I had written this patterned off of `os_version`, but this raises an interesting question to me: if this is optional, and this value is not always present, what should it be in those cases?

Does it make more sense to cast this value to a string so that we can support Unknown? 

It also strikes me that windows build numbers are not _really_ interpretable as numbers, i.e. they are more ordinal and as such string comparison is probably a better fit.

cc @badboy curious what you think as well

Reply:
You are right. Quantity is an integer ultimately and `"Unknown"` is not used anywhere in the code.
According to https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/data/environment.html it's a number in legacy, too.

There's relatively few cases where on Windows we would NOT be able to determine the value, right?
What does "Unknown" give us over just not setting the value (as the code does right now)?

Reply:
> What does "Unknown" give us over just not setting the value (as the code does right now)?

Slightly better conceptual alignment. Practically, nothing I suppose. Ok updating